src/main/kotlin/uk/gov/justice/digital/assessments/services/AssessmentService.kt

```diff
@@ -121,22 +124,29 @@ class AssessmentService(
         }
       }
     }
-
     return answers
   }
 
-  private fun createFromSupervision(supervisionId: String?): AssessmentDto {
-    val existingAssessment = assessmentRepository.findBySupervisionId(supervisionId)
-
-    if (existingAssessment != null) {
-      log.info("Existing assessment found for supervision $supervisionId")
-      return AssessmentDto.from(existingAssessment)
+  private fun createFromDelius(eventId: Int?, crn: String?, assessmentType: AssessmentType?): AssessmentDto {
+    if (eventId == null || crn.isNullOrEmpty() || assessmentType == null) {
+      throw IllegalStateException("Unable to create OASys Assessment with assessment type: $assessmentType, eventId: $eventId, crn: $crn")
     }
-
-    val newAssessment =
-      assessmentRepository.save(AssessmentEntity(supervisionId = supervisionId, createdDate = LocalDateTime.now()))
-    log.info("New assessment created for supervision $supervisionId")
-    return AssessmentDto.from(newAssessment)
+    val existingSubject = subjectRepository.findBySourceAndSourceIdAndCrn(deliusSource, eventId.toString(), crn)
+    if (existingSubject != null) {
+      log.info("Existing assessment ${existingSubject.assessment?.assessmentUuid} found for delius event id: $eventId, crn: $crn")
+      return AssessmentDto.from(existingSubject.assessment)
+    }
+    val offender = offenderService.getOffender(crn)
+    val oasysOffenderPk = assessmentUpdateRestClient.createOasysOffender(crn = crn, deliusEvent = eventId)
+    val oasysSetPK = oasysOffenderPk?.let { assessmentUpdateRestClient.createAssessment(it, assessmentType) }
```

Comment:
It fells like oasysOffenderPk being null should be an error condition. Or are we just not letting, say, an OASys outage stop us :)

Reply:
I actually followed the pattern for this function from the create assessment from court case where it also handles it as a nullable, I can't actually think how it would ever be null though as opposed to serving a 400 or 500

Reply:
Yep, that seems fair.